backend/services/interfaces/testSessionService.ts

```diff
@@ -43,13 +43,49 @@ export interface TestSessionResponseDTO {
    * the result of the test session
    * there should be one entry here per student
    * */
-  results?: [Result];
+  results?: ResultResponseDTO[];
   /** the code that students can use to access the test when it is live */
   accessCode: string;
   /** the time when the test session is started by teacher */
   startTime: Date;
 }
 
+export interface ResultRequestDTO {
```

Comment:
why do we add `ResultRequestDTO` and `ResultResponseDTO`? is it for the sake of modularity (since we are using it as a DTO) or is it strictly necessary for another reason?

Reply:
just to keep the request and response objects independent in case they need to diverge in the future

Reply:
makes sense, good call